File: exercise/python/day03/preparation/Gift.py
class Gift:
    def __init__(self, name: str, weight: float, color: str, material: str):
        self.name = name
        self.weight = weight
        self.color = color
        self.material = material
        self.attributes = {}

    def add_attribute(self, key: str, value: str):
        self.attributes[key] = value

    def get_recommended_age(self) -> int:
        return int(self.attributes.get("recommendedAge", 0))

    def __str__(self):
        return f"A {self.color}-colored {self.name} weighing {self.weight} kg made in {self.material}"

File: exercise/kotlin/day03/src/main/kotlin/preparation/Gift.kt
package preparation

class Gift(
    private val name: String,
    private val weight: Double,
    private val color: String,
    private val material: String
) {
    private val attributes: MutableMap<String, String> = mutableMapOf();

    fun addAttribute(key: String, value: String) {
        attributes[key] = value
    }

    fun getRecommendedAge(): Int = attributes["recommendedAge"]?.toIntOrNull() ?: 0

    override fun toString(): String = "A $color-colored $name weighing $weight kg made in $material"
}